src/coreclr/vm/dllimport.cpp

```diff
@@ -2272,15 +2272,7 @@ DWORD NDirectStubLinker::EmitProfilerBeginTransitionCallback(ILCodeStream* pcsEm
         EmitLoadStubContext(pcsEmit, dwStubFlags);
     }
 
-    if (SF_IsForwardStub(dwStubFlags))
-    {
-        pcsEmit->EmitLDLOC(GetThreadLocalNum());
-    }
-    else
-    {
-        // we use a null pThread to indicate reverse interop
-        pcsEmit->EmitLoadNullPtr();
-    }
+    pcsEmit->EmitLDLOC(GetThreadLocalNum());
```

Comment:
Will we always have a thread local? I like the explicit load of `null`, but I'm fine if we are guaranteed to always have a `null` initialized local. That means we also need to be clear that we aren't using `SkipLocalsInit` for the stub or its module.

Reply:
Yes, the thread local is created (and IL emitted to initialize it) on command when `GetThreadLocalNum()` is called.